OpenTranslator/Controllers/Awesome/AdminController.cs

```diff
@@ -280,7 +280,7 @@ public ActionResult Create()
 		[HttpPost]
 		public ActionResult Create(AdminInput input)
 		{
-			if (!ModelState.IsValid || this.doesTextIdExist(input) == true)
+			if (!ModelState.IsValid || this.doesTextIdExist(input) == true ||this.doesTextExist(input) == true)
```

Comment:
**1:** Method _doesTextIdExist_ is duplicated. this functionality already exists in TranslationRepository, it is called: _IsTextIdAlreadyStoraged(AdminInput input)_. Please change that validation for repository.IsTextIdAlreadyStoraged and delete the method doesTextIdExist.

**2:** If I am right, _doesTextExist_ validates if translated_text exists for new TextId, and if translated_text exists for existing TextId. **BUT**, there is another method on TranslationRepository called _IsTranslateAlreadyStoraged(AdminInput input, string translatedText)_ that validates if current translated_tex is already on db.
So, you can make a private method HERE, and call this two repository methods in order to make your validation more beautiful. It would be two or three lines, instead of the amount doesTextExist has.

**3:** methods in .Net should use PascalCase as name convention.

Reply:
I have Removed my function and used from Repository, It works fine for textId but for checking text duplication we have to change the method  or write new because we have to check only for text and language code but now that method consist textId as well. so If I insert record like
textId: test1 text: test languagecode:en. It will insert even if we have record for same text.like - textId: test2 text: test languagecode:en.
Hence I think we have to check for only text and language code.